### PR TITLE
Add Go Pipeline

### PR DESCRIPTION
**Pull request by kehoecj**
Adds a Go Pipeline using GitHub actions to run the following jobs on all branches:

- Static Analysis
- Go Formatting
- Running Unit Tests
- Building

Also fixes some formatting errors so the pipeline will pass. The pipeline can be viewed [here](https://github.com/kehoecj/config-file-validator/actions/runs/3696811596)

### CODE REVIEW

File: pkg/validator/validator_test.go [hunk 1]
```diff
@@ -5,33 +5,32 @@ import (
 )
 
 var testData = []struct {
-	name string
-	testInput []byte
+	name           string
```
Comment (jd4235) [inline pkg/validator/validator_test.go:8]: Does fmt align the variables like this or was this manual?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on: push
+
+jobs:
+
+  build:
```
Comment (jd4235) [inline .github/workflows/go.yml:10]: Nit-pick but this job name implies build only but analysis and test are included. Suggest changing job name to reflect activities or split out into multiple sequential jobs.

File: pkg/validator/validator_test.go (hunk 1 above)
Comment (kehoecj) [reply pkg/validator/validator_test.go:8]: fmt does it

File: .github/workflows/go.yml (hunk 2 above)
Comment (kehoecj) [reply .github/workflows/go.yml:10]: Good catch! I modified one of the templates and missed this. I'll look into your suggestions and see what aligns with the Gitlab Actions best practices.

File: pkg/validator/validator_test.go (hunk 1 above)
Comment (jackswiney) [reply pkg/validator/validator_test.go:8]: That's pretty gross. Is there a way to disable that specifically?

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+        
+    - name: Static Analysis
+      run: |
+        go vet ./...
```
[the remark this answers is not in the document]
Comment (kehoecj) [reply .github/workflows/go.yml:22]: Good catch! Will change

File: pkg/validator/validator_test.go [hunk 4]
```diff
@@ -5,33 +5,32 @@ import (
...
-			if ! valid && err == nil {
-				t.Error("incorrect result: function returned a nil error")
-			}
+		if valid && err != nil {
+			t.Error("incorrect result: err was not nil", err)
 		}
-}
\ No newline at end of file
+
+		if !valid && err == nil {
```
Comment (jackswiney) [inline pkg/validator/validator_test.go:32]: Is there a way to combine these with `if !valid || err != nil` and then just make the error a little more generic by formatting a string with valid and err

File: pkg/validator/validator_test.go (hunk 1 above)
Comment (kehoecj) [reply pkg/validator/validator_test.go:8]: I think it looks worse on the MR diff. Here's what the changes look like on the file itself:

image

I don't like so much indentation but that's the go standard and I'm hesitant to deviate from it.

Comment (jackswiney) [reply pkg/validator/validator_test.go:8]: Oh okay that's my bad, that looks completely fine.

File: pkg/validator/validator_test.go (hunk 4 above)
Comment (kehoecj) [reply pkg/validator/validator_test.go:32]: There's probably a better way to do it than is currently implemented but combining them into `if !valid || err != nil` doesn't quite test the same thing. The first if statement checks that an error is not returned if the function returns true and the second ensures that an error is returned if the function returns false. The test should still pass if the function returns false if that's the expected result.

Comment (jackswiney) [reply pkg/validator/validator_test.go:32]: Yeah sorry that should be an xor but I just realized go doesn't have that builtin so it wouldn't make it prettier anyways. Oh well.

**Review by jackswiney — APPROVED** [review]
Resolved my comments, lgtm

**Pull request merged**